crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -418,59 +425,62 @@ where
+
+        // This is a bit risky because it's possible to change something that would
+        // break the prepass but be fine in the main pass.
+        // Since this api is pretty low-level it doesn't matter that much, but it is a potential issue.
+        M::specialize(&self.material_pipeline, &mut descriptor, layout, key)?;
+
+        Ok(descriptor)
+    }
+}
+
+impl PrepassPipelineInternal {
+    fn specialize(
+        &self,
+        mesh_key: MeshPipelineKey,
+        shader_defs: Vec<ShaderDefVal>,
+        layout: &MeshVertexBufferLayoutRef,
+    ) -> Result<RenderPipelineDescriptor, SpecializedMeshPipelineError> {
+        let mut shader_defs = shader_defs;
+        let mut bind_group_layouts = vec![if mesh_key
             .contains(MeshPipelineKey::MOTION_VECTOR_PREPASS)
         {
             self.view_layout_motion_vectors.clone()
         } else {
             self.view_layout_no_motion_vectors.clone()
         }];
-        let mut shader_defs = Vec::new();
         let mut vertex_attributes = Vec::new();
-
-        // Let the shader code know that it's running in a prepass pipeline.
-        // (PBR code will use this to detect that it's running in deferred mode,
-        // since that's the only time it gets called from a prepass pipeline.)
         shader_defs.push("PREPASS_PIPELINE".into());
-
-        // NOTE: Eventually, it would be nice to only add this when the shaders are overloaded by the Material.
-        // The main limitation right now is that bind group order is hardcoded in shaders.
         bind_group_layouts.push(self.material_layout.clone());
-
         #[cfg(all(feature = "webgl", target_arch = "wasm32", not(feature = "webgpu")))]
         shader_defs.push("WEBGL2".into());
-
         shader_defs.push("VERTEX_OUTPUT_INSTANCE_INDEX".into());
-
-        if key.mesh_key.contains(MeshPipelineKey::DEPTH_PREPASS) {
+        if mesh_key.contains(MeshPipelineKey::DEPTH_PREPASS) {
             shader_defs.push("DEPTH_PREPASS".into());
         }
-
-        if key.mesh_key.contains(MeshPipelineKey::MAY_DISCARD) {
+        if mesh_key.contains(MeshPipelineKey::MAY_DISCARD) {
             shader_defs.push("MAY_DISCARD".into());
         }
-
-        let blend_key = key
-            .mesh_key
-            .intersection(MeshPipelineKey::BLEND_RESERVED_BITS);
+        let blend_key = mesh_key.intersection(MeshPipelineKey::BLEND_RESERVED_BITS);
         if blend_key == MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA {
             shader_defs.push("BLEND_PREMULTIPLIED_ALPHA".into());
         }
         if blend_key == MeshPipelineKey::BLEND_ALPHA {
             shader_defs.push("BLEND_ALPHA".into());
         }
-
         if layout.0.contains(Mesh::ATTRIBUTE_POSITION) {
             shader_defs.push("VERTEX_POSITIONS".into());
             vertex_attributes.push(Mesh::ATTRIBUTE_POSITION.at_shader_location(0));
         }
-
-        // For directional light shadow map views, use unclipped depth via either the native GPU feature,
```

Comment:
Was removing all of these comments intentional?

Reply:
Nope, not sure how that happened, I'll put them back